src/libmongoc/tests/unified/result.c

```diff
@@ -150,6 +150,11 @@ result_from_bulk_write (result_t *result, bson_t *reply, bson_error_t *error)
          rewritten_upserted_ids = rewrite_upserted_ids (upserted_ids);
          BSON_APPEND_DOCUMENT (
             write_result, "upsertedIds", rewritten_upserted_ids);
+      } else {
+         /* upsertedIds is a required field in BulkWriteResult, so append an
+          * empty document even if not documents were upserted. */
+         upserted_ids = bson_new ();
+         BSON_APPEND_DOCUMENT (write_result, "upsertedIds", upserted_ids);
```

Comment:
This appears to have been an oversight in the original test runner implementation.